Simple-Projects/Simple_Projects/Unit/Unit.cpp

```diff
@@ -31,8 +32,9 @@ const std::string& Unit::getName() const {
 void Unit::addHitPoints(int hp) {
     if (hitPoints + hp > hitPointsLimit) {
         hitPoints = hitPointsLimit;
+    } else {
+        hitPoints += hp;
```

Comment:
Bug-fix: the second addition should be in an 'else' branch. Otherwise HP overflow may occur.